modules/core/client/components/ReadMorePanel.js

```diff
@@ -0,0 +1,71 @@
+// External dependencies
+import PropTypes from 'prop-types';
+import React, { Component } from 'react';
+
+// Internal dependencies
+import { plainText, plainTextLength } from '../utils/filters';
+
+export class ReadMorePanel extends Component {
+  constructor(props) {
+    super(props);
+    this.toggleMore = this.toggleMore.bind(this);
+    this.state = {
+      showMore: false
+    };
+  }
+
+  toggleMore() {
+    this.setState({ showMore: true });
+  }
+
+  componentDidCatch(error, errorInfo) {
+    // You can also log the error to an error reporting service
+    console.error(error, errorInfo); //eslint-disable-line
+  }
+
+  render() {
+    const { showMore } = this.state;
+    const { content, id, limit } = this.props;
+
+    if (content.length === 0) {
+      return;
+    }
+
+    if (showMore || plainTextLength(content) <= limit) {
+      return <div dangerouslySetInnerHTML={{ __html: content }} />;
+    }
+
+    return (
+      <div className="panel-more-wrap">
+        <div
+          className="panel-more-wrap"
+          dangerouslySetInnerHTML={{ __html: plainText(content).substr(0, limit) }}
+          id={id}
+          onClick={this.toggleMore}
+        />
+        <div
+          aria-controls={id}
+          aria-expanded="false"
+          className="panel-more-fade"
+          onClick={this.toggleMore}
+          type="button"
+        >
+          {/* TODO: translate */}
```

Comment:
I actually added `withTranslate` here but it was failing with `t is not function`. Will have to look tomorrow again with fresh eyes.

Reply:
@mrkvon haha, I'm again with sleepy eyes. I pushed the translation code now and still getting `t is not a function` — can you spot what I'm doing wrong?

Reply:
Yay, this started working in https://github.com/Trustroots/trustroots/pull/1161/commits/662494a65c2a0033e1b0f83e8b1079a186481104.